### PR TITLE
Routing: Redirect lowercase urls

### LINKED ISSUE #541: Collections: lower case URLs just get marked no-index instead of canonicalised
Pages with lower case URL suffix, like, https://www.ifixit.com/Parts/iphone, https://www.ifixit.com/Parts/ipad, are just no-indexed. I think we had planned to canonicalise them? 

I don't know where we landed on desired case sensitivity but I don't think this was the desired behavior. 

https://docs.google.com/document/d/15bJZoTv3Sh7-ezn8cd6P-WzsnL5HQIk59o8n3NO3-Hc/edit

@sterlinghirsh

### PR DESCRIPTION
**Pull request by masonmcelvain**
- Uses the new [`eqi` filter](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest/filtering-locale-publication.html#filtering) ([introduced](https://github.com/strapi/strapi/releases) in release 4.3.0) to lookup product lists in a case insensitive manner.

## QA

Lower case urls should get redirected (with a 308 permanent redirect) to the canonical page.

CC @sterlinghirsh 

Closes #541

### CODE REVIEW

**Comment by masonmcelvain** [comment]
dev_block ✌🏻 on #555 merging first

**Comment by sterlinghirsh** [comment]
When the canonical capitalization is a different case than what was requested, we should redirect rather than loading the page with the wrong url.

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻

File: frontend/models/product-list/index.ts [hunk 1]
```diff
@@ -48,20 +48,21 @@ export async function findProductList(
    filters: ProductListFiltersInput,
    deviceItemType: string | null = null
 ): Promise<ProductList | null> {
-   const deviceTitle = filters.deviceTitle?.eq ?? '';
+   const filterDeviceTitle = filters.deviceTitle?.eq ?? '';
```
Comment (sterlinghirsh) [inline frontend/models/product-list/index.ts:51]: This just changed right?
```suggestion
   const filterDeviceTitle = filters.deviceTitle?.eqi ?? '';
```

File: frontend/pages/Parts/index.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ export const getServerSideProps: GetServerSideProps<AppPageProps> = async (
          getGlobalSettings(),
          getStoreList(),
          getStoreByCode('us'),
-         findProductList({ handle: { eq: 'Parts' } }),
+         findProductList({ handle: { eqi: 'Parts' } }),
```
Comment (sterlinghirsh) [inline frontend/pages/Parts/index.tsx:36]: Not sure it matters but I think this one didn't have to change.

File: frontend/pages/Parts/[...deviceHandleItemType].tsx [hunk 3]
```diff
@@ -75,6 +76,15 @@ export const getServerSideProps: GetServerSideProps<AppPageProps> = async (
       };
    }
 
+   if (productList.deviceTitle && productList.deviceTitle !== deviceTitle) {
+      return {
+         redirect: {
+            permanent: true,
+            destination: `/Parts/${encodeDeviceTitle(productList.deviceTitle)}`,
```
Comment (sterlinghirsh) [inline frontend/pages/Parts/[...deviceHandleItemType].tsx:83]: Might need to account for item type here too.

File: frontend/pages/Tools/[handle].tsx [hunk 4]
```diff
@@ -50,6 +51,15 @@ export const getServerSideProps: GetServerSideProps<AppPageProps> = async (
       };
    }
 
+   if (productList.handle && productList.handle !== handle) {
+      return {
+         redirect: {
+            permanent: true,
+            destination: `/Tools/${encodeDeviceTitle(productList.handle)}`,
```
Comment (sterlinghirsh) [inline frontend/pages/Tools/[handle].tsx:58]: I think the productList handle comes pre-encoded 
image

**Comment by sterlinghirsh** [comment]
CR :zap: dev_block :zap: on a few small things but thanks for taking this on

File: frontend/pages/Tools/[handle].tsx (hunk 4 above)
Comment (masonmcelvain) [reply frontend/pages/Tools/[handle].tsx:58]: Cool. Looking at `marketing` types, I'm gonna assume they come pre-encoded too.

File: frontend/models/product-list/index.ts (hunk 1 above)
Comment (masonmcelvain) [reply frontend/models/product-list/index.ts:51]: good catch thanks

File: frontend/pages/Tools/[handle].tsx (hunk 4 above)
Comment (sterlinghirsh) [reply frontend/pages/Tools/[handle].tsx:58]: yes

File: frontend/pages/Parts/index.tsx (hunk 2 above)
Comment (masonmcelvain) [reply frontend/pages/Parts/index.tsx]: https://github.com/iFixit/react-commerce/pull/556/commits/0657ba2256fd53073aa3f547ba4dd6a4aa597384

File: frontend/pages/Tools/[handle].tsx (hunk 4 above)
Comment (masonmcelvain) [reply frontend/pages/Tools/[handle].tsx]: https://github.com/iFixit/react-commerce/pull/556/commits/f8c35cdd69403d62e810146fe6bf7ca5fa0a2141

File: frontend/pages/Parts/[...deviceHandleItemType].tsx (hunk 3 above)
Comment (masonmcelvain) [reply frontend/pages/Parts/[...deviceHandleItemType].tsx]: https://github.com/iFixit/react-commerce/pull/556/commits/08d1ad5343fdbf38960a67d5dd3be934bf67d247

**Comment by masonmcelvain** [comment]
Thanks @sterlinghirsh appreciate the feedback. un_dev_block ✌🏻

**Comment by sterlinghirsh** [comment]
CR :zap:

**Review by sterlinghirsh — APPROVED** [review]

**Comment by deltuh-vee** [comment]
QA 🎬 
Lowercase urls are now redirected with a 308 Status code.
❓ Are Item Types also supposed to be redirected to a different canonical URL?
CleanShot 2022-08-02 at 14 02 22

**Comment by masonmcelvain** [comment]
> Are Item Types also supposed to be redirected to a different canonical URL?

I didn't do that in this PR. Nothing in the [doc](https://docs.google.com/document/d/15bJZoTv3Sh7-ezn8cd6P-WzsnL5HQIk59o8n3NO3-Hc/edit) indicated that we wanted to set up redirects for item types. If that's what we want, let's make a new issue.

**Pull request merged**